src/main/java/com/livevox/challenge/app/vendor/VendorClient.java

```diff
@@ -0,0 +1,78 @@
+package com.livevox.challenge.app.vendor;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+@Service
+@Slf4j
+public class VendorClient {
+
+    private static final String BASE_URL = "https://external-api-challenge-2022.herokuapp.com/api/";
+    private static final String CALLS_URL = String.format("%s%s", BASE_URL, "calls/daily");
+    private static final String COUNTRIES_URL = String.format("%s%s", BASE_URL, "countries/{prefix}");
+    private final Map<Integer, Set<CountryResponse.Country>> countries;
+    private final RestTemplate restTemplate;
+
+    public VendorClient(final RestTemplateBuilder builder) {
+        this.restTemplate = builder.build();
+        this.countries = new HashMap<>();
+    }
+
+    public RemoteCallResponse fetchCalls() {
+        final ResponseEntity<RemoteCallResponse> response = getResponse(CALLS_URL, RemoteCallResponse.class);
+        final RemoteCallResponse defValue = RemoteCallResponse.builder().calls(Collections.emptyList()).build();
+        return handleResponse(response, defValue);
+    }
+
+    public Set<CountryResponse.Country> fetchCountriesByPrefix(final int prefix) {
```

Comment:
I guess it would be ideal to try `@Cacheable` here so that it would be easier to read this.